src/tiled/wangdock.cpp

```diff
@@ -284,6 +294,14 @@ void WangDock::eraseWangIdsButtonClicked()
     emit currentWangIdChanged(mCurrentWangId);
 }
 
+void WangDock::switchTemplateViewButtonClicked()
+{
+    if (mWangTemplateView->isVisible())
+        setColorView(true);
+    else
+        setTemplateView(true);
```

Comment:
This way of switching appears to behave a little strange (with the height changing upon switching). I would suggest to use a QStackedWidget (or QStackedLayout if possible) to switch between the two views.